sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -36,4 +56,145 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
       })
       .getMany();
   }
+
+  /**
+   * Save scholastic standing and create new assessment
+   * for an application.
+   * @param applicationId application id.
+   * @param scholasticStandingData scholastic standing data to be saved
+   */
+  async saveScholasticStandingCreateReassessment(
+    applicationId: number,
+    userId: number,
+    scholasticStandingData: ScholasticStandingAPIInDTO,
+  ): Promise<StudentScholasticStanding> {
+    const application = await this.applicationRepo
+      .createQueryBuilder("application")
+      .select(["application", "currentAssessment.id", "offering.id"])
+      .where("application.id = :applicationId", { applicationId })
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .getOne();
+
+    if (!application) {
+      throw new CustomNamedError(
+        "Application Not found.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (application.applicationStatus !== ApplicationStatus.completed) {
+      throw new CustomNamedError(
+        "Not a completed application.",
+        NOT_A_COMPLETED_APPLICATION,
+      );
+    }
+
+    return this.connection.transaction(async (transactionalEntityManager) => {
+      const now = new Date();
+      const auditUser = {
```

Comment:
I do not see a reason to break this small object into 3 lines. Please consider keeping it in just a single line (ESLint allows it).
```typescript
const auditUser = { id: userId } as User;
```